### PR TITLE
TMA-787 added support M:N in LCM

### PR DESCRIPTION
**Pull request by Hahihula**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/438/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1269/ : FAILURE in 2m 48s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1539/ : FAILURE in 2m 32s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 1]
```diff
@@ -69,6 +69,71 @@
   end
 end
 
+describe 'The date dimension stuff' do
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:11, m_to_n_spec.rb:72 (mass = 240)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 2]
```diff
@@ -81,5 +146,16 @@ def m_to_ns(project)
       []
     end
   end
-  result
+  if !model['projectModelView']['model']['projectModel']['dateDimensions'].nil?
+    date_dim_result = model['projectModelView']['model']['projectModel']['dateDimensions'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:142, m_to_n_spec.rb:150 (mass = 68)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'd' (UncommunicativeVariableName)

File: lib/gooddata/models/blueprint/to_wire.rb [hunk 3]
```diff
@@ -109,12 +109,13 @@ def self.dataset_to_wire(project, dataset)
       # @param project [Hash] Project blueprint hash represenation
       # @param dataset [Hash] Dataset blueprint hash represenation
       # @return [Hash] Manifest for a particular reference
-      def self.date_dimension_to_wire(_project, dataset)
+      def self.date_dimension_to_wire(project, dataset)
```
Comment (yenkins) [inline lib/gooddata/models/blueprint/to_wire.rb]: Has approx 7 statements (TooManyStatements)

File: lib/gooddata/models/from_wire.rb [hunk 4]
```diff
@@ -165,7 +166,8 @@ def self.parse_references(dataset)
       # @param dataset [Hash] Whatever comes from wire
       # @return [Hash] Manifest for a particular bridge
       def self.parse_bridges(dataset)
-        references = dataset['dataset']['bridges'] || []
+        references = !dataset['dataset'].nil? && !dataset['dataset']['bridges'].nil? ? dataset['dataset']['bridges'] : []
```
Comment (yenkins) [inline lib/gooddata/models/from_wire.rb]: Calls 'dataset['dataset']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/from_wire.rb]: Calls 'dataset['dataset']['bridges']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/from_wire.rb]: Performs a nil-check (NilCheck)

File: lib/gooddata/models/from_wire.rb [hunk 5]
```diff
@@ -165,7 +166,8 @@ def self.parse_references(dataset)
...
+        references = !dataset['dateDimension'].nil? && !dataset['dateDimension']['bridges'].nil? ? dataset['dateDimension']['bridges'] : references
```
Comment (yenkins) [inline lib/gooddata/models/from_wire.rb]: Calls 'dataset['datedimension']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/from_wire.rb]: Calls 'dataset['datedimension']['bridges']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 6]
```diff
@@ -81,5 +146,16 @@ def m_to_ns(project)
...
+      if !d['dateDimension']['bridges'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']['bridges']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 7]
```diff
@@ -81,5 +146,16 @@ def m_to_ns(project)
       []
     end
   end
-  result
+  if !model['projectModelView']['model']['projectModel']['dateDimensions'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['projectmodelview']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['projectmodelview']['model']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['projectmodelview']['model']['projectmodel']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['projectmodelview']['model']['projectmodel']['datedimensions']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 8]
```diff
@@ -69,6 +69,71 @@
...
+
+    project_helper = Support::ProjectHelper.create(
+      client: @rest_client,
+      title: "Development Project for m to n #{@suffix}",
+      auth_token: @config[:dev_token],
+      environment: @config[:environment],
+      ads: @ads
+    )
+
+    @dateDimProject = project_helper.project
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Style/VariableName: Use snake_case for variable names.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/439/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1270/ : FAILURE in 1m 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1540/ : FAILURE in 1m 00s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 9]
```diff
@@ -64,6 +64,40 @@
     expect(m_to_ns(@project)).to include(['dataset.y', 'dataset.x'])
   end
 
+  it 'on datedimension too' do
+    project_helper = Support::ProjectHelper.create(
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Identical code found in m_to_n_spec.rb:31, m_to_n_spec.rb:68 (mass = 84)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 10]
```diff
@@ -81,5 +115,16 @@ def m_to_ns(project)
       []
     end
   end
-  result
+  if !model['projectModelView']['model']['projectModel']['dateDimensions'].nil?
+    date_dim_result = model['projectModelView']['model']['projectModel']['dateDimensions'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:111, m_to_n_spec.rb:119 (mass = 68)

File: spec/unit/models/blueprint/to_wire_spec.rb [hunk 11]
```diff
@@ -192,7 +192,7 @@
   describe '#date_dimension_to_wire' do
     it 'should be able to generate date dimension' do
       res = ToWire.date_dimension_to_wire(@spec, type: "date_dimension", id: "timeline", title: "Timeline")
-      expect(res).to eq(:dateDimension => { :name => "timeline", :title => "Timeline" })
+      expect(res).to eq(:dateDimension => { :name => "timeline", :title => "Timeline", :bridges => []})
```
Comment (yenkins) [inline spec/unit/models/blueprint/to_wire_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside } missing.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/444/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1275/ : FAILURE in 1m 28s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1545/ : SUCCESS in 1m 03s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 12]
```diff
@@ -73,13 +107,24 @@ def m_to_ns(project)
   options = { include_ca: true }
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  result = model['datasets'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:111, m_to_n_spec.rb:119 (mass = 56)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'd' (UncommunicativeVariableName)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 13]
```diff
@@ -73,13 +107,24 @@ def m_to_ns(project)
...
     if !d['dataset']['bridges'].nil?
       [d['dataset']['identifier'], d['dataset']['bridges'].join]
     else
       []
     end
   end
-  result
+  if !model['dateDimensions'].nil?
+    date_dim_result = model['dateDimensions'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:111, m_to_n_spec.rb:119 (mass = 56)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 14]
```diff
@@ -73,13 +107,24 @@ def m_to_ns(project)
...
+      if !d['dateDimension']['bridges'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']['bridges']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 15]
```diff
@@ -73,13 +107,24 @@ def m_to_ns(project)
...
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  result = model['datasets'].map do |d|
     if !d['dataset']['bridges'].nil?
       [d['dataset']['identifier'], d['dataset']['bridges'].join]
     else
       []
     end
   end
-  result
+  if !model['dateDimensions'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['datedimensions']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 16]
```diff
@@ -73,13 +107,24 @@ def m_to_ns(project)
   options = { include_ca: true }
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'project.client' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/445/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1276/ : FAILURE in 1m 29s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1546/ : SUCCESS in 1m 01s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 17]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
...
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  result = model['datasets'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:111, m_to_n_spec.rb:115 (mass = 56)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'd' (UncommunicativeVariableName)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 18]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
...
+    !d['dataset']['bridges'].nil? ? [d['dataset']['identifier'], d['dataset']['bridges'].join] : []
+  end
+  if !model['dateDimensions'].nil?
+    date_dim_result = model['dateDimensions'].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Similar code found in m_to_n_spec.rb:111, m_to_n_spec.rb:115 (mass = 56)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 19]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
...
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  result = model['datasets'].map do |d|
+    !d['dataset']['bridges'].nil? ? [d['dataset']['identifier'], d['dataset']['bridges'].join] : []
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['dataset']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['dataset']['bridges']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 20]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
...
+  end
+  if !model['dateDimensions'].nil?
+    date_dim_result = model['dateDimensions'].map do |d|
+      if !d['dateDimension']['bridges'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd['datedimension']['bridges']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 21]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
...
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  result = model['datasets'].map do |d|
+    !d['dataset']['bridges'].nil? ? [d['dataset']['identifier'], d['dataset']['bridges'].join] : []
+  end
+  if !model['dateDimensions'].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model['datedimensions']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 22]
```diff
@@ -73,13 +107,20 @@ def m_to_ns(project)
   options = { include_ca: true }
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'project.client' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/446/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1277/ : FAILURE in 1m 26s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1547/ : SUCCESS in 1m 09s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 23]
```diff
@@ -73,13 +107,17 @@ def m_to_ns(project)
...
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    if !model[a].nil?
+      model[a].map do |d|
+        !d[a.chomp('s')]['bridges'].nil? ? [d[a.chomp('s')]['identifier'], d[a.chomp('s')]['bridges'].join] : []
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'a.chomp('s')' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd[a.chomp('s')]' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd[a.chomp('s')]['bridges']' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 24]
```diff
@@ -73,13 +107,17 @@ def m_to_ns(project)
...
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    if !model[a].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model[a]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 25]
```diff
@@ -73,13 +107,17 @@ def m_to_ns(project)
   options = { include_ca: true }
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'project.client' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 26]
```diff
@@ -73,13 +107,17 @@ def m_to_ns(project)
...
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    if !model[a].nil?
+      model[a].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'd' (UncommunicativeVariableName)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 27]
```diff
@@ -73,13 +107,17 @@ def m_to_ns(project)
...
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'a' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/447/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1278/ : FAILURE in 1m 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1548/ : SUCCESS in 1m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/448/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1279/ : SUCCESS in 1m 00s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1549/ : SUCCESS in 1m 00s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 28]
```diff
@@ -73,13 +101,17 @@ def m_to_ns(project)
...
+    if model[a].nil?
       []
+    else
+      model[a].map do |d|
+        d[a.chomp('s')]['bridges'].nil? ? [] : [d[a.chomp('s')]['identifier'], d[a.chomp('s')]['bridges'].join]
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'a.chomp('s')' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd[a.chomp('s')]' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'd[a.chomp('s')]['bridges']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Style/TrailingWhitespace: Trailing whitespace detected.

File: spec/lcm/integration/m_to_n_spec.rb [hunk 29]
```diff
@@ -73,13 +101,17 @@ def m_to_ns(project)
...
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    if model[a].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model[a]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 30]
```diff
@@ -73,13 +101,17 @@ def m_to_ns(project)
   options = { include_ca: true }
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'project.client' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 31]
```diff
@@ -73,13 +101,17 @@ def m_to_ns(project)
...
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    if model[a].nil?
       []
+    else
+      model[a].map do |d|
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Has the variable name 'd' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/449/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1280/ : FAILURE in 1m 08s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1550/ : SUCCESS in 1m 13s

File: spec/lcm/integration/m_to_n_spec.rb [hunk 32]
```diff
@@ -73,13 +101,14 @@ def m_to_ns(project)
...
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
-    end
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
+  %w(
+    datasets
+    dateDimensions
+  ).map do |a|
+    return [] if model[a].nil?
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'model[a]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Performs a nil-check (NilCheck)

File: spec/lcm/integration/m_to_n_spec.rb [hunk 33]
```diff
@@ -73,13 +101,14 @@ def m_to_ns(project)
...
   result = project.client.get("/gdc/projects/#{project.pid}/model/view", params: { includeDeprecated: true, includeGrain: true, includeCA: options[:include_ca] })
   polling_url = result['asyncTask']['link']['poll']
-  model = project.client.poll_on_code(polling_url, options)
-  result = model['projectModelView']['model']['projectModel']['datasets'].map do |d|
-    if !d['dataset']['bridges'].nil?
-      [d['dataset']['identifier'], d['dataset']['bridges'].join]
-    else
-      []
-    end
+  model = project.client.poll_on_code(polling_url, options)['projectModelView']['model']['projectModel']
```
Comment (yenkins) [inline spec/lcm/integration/m_to_n_spec.rb]: Calls 'project.client' 2 times (DuplicateMethodCall)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/450/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1281/ : SUCCESS in 1m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1551/ : SUCCESS in 1m 08s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/451/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1282/ : SUCCESS in 1m 05s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1552/ : SUCCESS in 1m 07s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/112/ : SUCCESS in 18s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1283/ : SUCCESS in 1m 18s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1553/ : SUCCESS in 1m 10s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1484/ : SUCCESS in 14m 27s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/56/ : SUCCESS in 38m 32s

**Pull request merged**